src/modules/job-list/list.c

```diff
@@ -159,6 +159,93 @@ json_t *get_jobs (struct job_state_ctx *jsctx,
     return NULL;
 }
 
+static int legacy_list_rpc (flux_t *h,
```

Comment:
FYI - I did test this scenario (run the new flux as an instance in an older flux) and `flux jobs` and `flux pstree` as examples work correctly, so nice work!

Reply:
ahhh, covered both cases in updated comment

```
 job-list: support legacy RPC protocol
    
    Problem: It would not be unreasonable for a user to launch a flux
    version that is different than the parent version of flux (especially
    when then parent version is the system instance).  With the recent
    update of the job list module to use "constraints", the older RPC
    format would not allow a user to (for example) run:
    
    flux --parent jobs
    
    If the parent version is a newer version of flux.  Or run:
    
    flux jobs -R
    
    if the subinstance is a newer version of flux.
    
    Solution: Support the older RPC protocol in job-list.  Internally,
    simply create a constraint object given the old protocol's inputs.
    
    Fixes #5363


```